.github/workflows/CI.yml

```
@@ -1,13 +1,20 @@
 name: Rust CI
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```

Triggering the workflow on both push and pull_request events makes it run twice unnecessarily, when a PR is open, and also when it is merged. Additionally, all PRs, even when not directed to main, should be subject to these checks.

```suggestion
on: [pull_request]
```

---

Okay thanks let me change that